consul/tests/test_integration.py

```diff
@@ -98,6 +98,12 @@ def test_prometheus_endpoint(aggregator, dd_environment, instance_prometheus, ca
             aggregator.assert_metric(metric, tags=common_tags, count=1)
 
         aggregator.assert_metric('consul.peers', value=3, count=1)
+        aggregator.assert_metric_has_tag_prefix('consul.raft.replication.appendEntries.logs', 'peer_id', count=2)
+
+        for hist_suffix in ['count', 'sum', 'quantile']:
+            aggregator.assert_metric_has_tag('consul.http.request.{}'.format(hist_suffix), 'method:GET', at_least=0)
+            for metric in common.PROMETHEUS_HIST_METRICS:
+                aggregator.assert_metric_has_tag(metric + hist_suffix, 'foo:bar', at_least=1)
```

Comment:
? More rigorously we probably actually want to check everything in `common_tags`:

```suggestion
                for tag in common_tags:
                    aggregator.assert_metric_has_tag(metric + hist_suffix, 'foo:bar', at_least=1)
```